src/security/PeerConnector.cc

```diff
@@ -278,9 +278,9 @@ Security::PeerConnector::handleNegotiationResult(const Security::IoResult &resul
     }
 
     // TODO: Honor result.important when working in a reverse proxy role?
-    debugs(83, 2, "ERROR: Cannot establish a TLS connection to " << serverConnection() << ':' <<
-           Debug::Extra << "problem: " << result.errorDescription <<
```

Comment:
So here we have the actual bug. This line should be displaying `result`, not just the `result.errorDescription` member.

None of the rest of this PR is relevant to fixing the bug, it is all just code bloat and obfuscation.

Reply:
> So here we have the actual bug. 

Yes, for some definition of an "actual bug". IMO, there are actually several relevant bugs, and none of them are confined to this code line.

> This line should be displaying `result`, not just the `result.errorDescription` member.

This line should not display `result` as-is (in the official code) because doing so will produce difficult-to-grok high-level cache.log output -- existing Security::IoResult::print() is not meant to be used as-is in such contexts. Doing so will also make code duplication worse. This PR refactors Security::IoResult::print() to make simple result printing usable here (among other changes such as reducing code duplication).

> None of the rest of this PR is relevant to fixing the bug, it is all just code bloat and obfuscation.

This PR reduces the probability of a very similar "actual bug" from occurring: The other two debugs() statements have the same buggy code that will trigger the same crashes if seemingly unrelated code is added or changed. The existing buggy debugs() statements are also likely to be copied to other contexts as we upgrade TLS code, increasing the bug surface.

More consistent, safer, and simpler high-level code is _not_ "code bloat or obfuscation". This PR is an improvement even if there were no underlying "real bugs" to fix here! If you can achieve better results with less code, please do so (or suggest specific PR changes to arrive at that state instead). Otherwise, please remove your negative vote.

P.S. WithExtras() addition should be evaluated for "code bloat". I have already done that in this PR description (there are more details in branch commit 518d1307) and concluded that this addition is warranted. No flaws in that logic have been identified so far.